### DISCUSSION TITLE
How can I secure the connection string in quartz.config

### DISCUSSION THREAD

**Discussion by norcino** [quartznet/quartznet#1296] [Q&A] [answered]
I would like to secure my connection string stored in the quartz.config, how can I achieve it?
I was thinking to encrypt it using _System.Security.Encryption.ProtectedData_ this means that I need to customize the way quartz get the connection strings to add and intermediate step to decrypt it.

Can anyone point me to the right direction?

**Comment by lahma** [comment, marked as answer]
Quartz can also resolve environment variables if that would help in your case, eventually you need some place to physically store the connection string, protected data is an option of course but there's no direct support. During all the years Quartz has been deployed I think this is actually the first time someone asks for such feature - nothing out of the box I guess.

**Reply by norcino** [reply]
Thanks for the reply Marko.
In the previous version of Quartz it was possible to redefine the Connection Provider type, see: https://stackoverflow.com/questions/50172365/can-quartzserver-net-decrypts-connection-string-in-config-file.
Is there any other way for example to configure a custom DBProvider?

**Reply by lahma** [reply]
I think that should still work, have you tried it or is there a blocker?

**Reply by norcino** [reply]
@lahma I was able to spend more time on this issue and I can confirm that it works.
The documentation although does not contain the **'connectionProvider.type'** which I needed to use:
https://www.quartz-scheduler.net/documentation/quartz-3.x/configuration/reference.html#datasources-ado-net-jobstores

```
quartz.jobStore.dataSource = defaultDS
quartz.dataSource.defaultDS.connectionProvider.type = QuartzDbProvider.EncryptedConnectionDbProvider, EncryptedConnectionDbProvider
```

Thank you

**Reply by lahma** [reply]
@norcino I've added the property to reference and also updated strong-typed configuration to support defining the connection provider:  https://github.com/quartznet/quartznet/pull/1312/files .

**Reply by norcino** [reply]
Well done thanks.

**Comment by norcino** [comment]
Being using the version 3 I did not even try as the documentation doesn't
have these configuration settings anymore, today I will try to go again
through the source code to find if there is anything I can hook into.
